lib/manageiq/automation_engine/logger.rb

```diff
@@ -1,31 +1,32 @@
 module ManageIQ
   module AutomationEngine
     class Logger < ManageIQ::Loggers::Base
-      attr_reader :automation_log_wrapper
+      attr_reader :automation_log_wrapper, :resource_id
 
-      def initialize(*args, automation_log_wrapper:, **kwargs)
+      def initialize(*args, automation_log_wrapper:, resource_id: nil, **kwargs)
         @automation_log_wrapper = automation_log_wrapper
+        @resource_id            = resource_id
+
         super(*args, **kwargs)
       end
 
-      def self.create_log_wrapper(io = File::NULL)
+      def self.create_log_wrapper(io: File::NULL, automation_log_wrapper: nil, resource_id: nil)
         # We modify the interface of logger methods such as info/warn/etc. to allow the keyword argument
         # resource_id. Therefore, we need to wrap all client logger calls to these methods to process the resource_id,
         # cut the request_log entry and forward the remaining arguments to the logger.
-        new(io, :progname => "automation", :automation_log_wrapper => Vmdb::Loggers.create_logger("automation.log"))
+        new(io, :progname => "automation", :automation_log_wrapper => automation_log_wrapper || Vmdb::Loggers.create_logger("automation.log"), :resource_id => resource_id)
       end
 
-      private def add_to_db(severity, message = nil, progname = nil, resource_id: nil)
-        return [severity, message, progname] unless resource_id
+      private
+
+      def add(severity, message = nil, progname = nil, &block)
+        automation_log_wrapper.add(severity, message, progname, &block)
```

Comment:
think this wants to be after we resolve `message`?

Doesn't seem optimal if we `message ||= yield` and then have `automation_log_wrapper` have to do the same thing

Reply:
I was thinking of the wrapped logger had a different progname that it wanted to use this would allow that

Reply:
Plus I think this matches the existing behavior because it used to call the wrapped logger directly from `def info` as opposed to from `add`